### ISSUE TITLE
Third-party matching trial token loaded in top level site is still not allowing 3P cookies access

### ISSUE THREAD

**Issue by annamalai87** [privacysandbox/privacy-sandbox-dev-support#330]
Our usecase(http://b/332975135) to access third party cookies for the origin https://pay.google.com is recently approved.
We could successfully inject the trial token with third party matching in the top level site when it loads our script. Although developer tool shows the token is valid in the application tab, when the top level site loads the iframe with the origin pay.google.com is not having the 3P cookies.

Here are the screenshots:

Top level site shows the trial token is injected:

![image](https://github.com/privacysandbox/privacy-sandbox-dev-support/assets/41874936/62eede6c-a585-45fb-b386-263c552fee50)

Developer tool shows the trial token is valid:

![image](https://github.com/privacysandbox/privacy-sandbox-dev-support/assets/41874936/94563a51-3576-40d7-9278-05430563b5fa)

Iframe with the origin pay.google.com still unable to access 3P cookies:

![image](https://github.com/privacysandbox/privacy-sandbox-dev-support/assets/41874936/cbdf1378-d422-4293-b91b-14768f3ea8ca)

Same Iframe when we enable the option to access 3P cookies manually can access 3P cookies:

![image](https://github.com/privacysandbox/privacy-sandbox-dev-support/assets/41874936/8068d2e2-4908-4fe4-82a0-98ccd4ec2c91)

**Comment by hoodjoshua** [comment]
Can you confirm that you have the following flag configuration?
```
chrome://flags/#third-party-cookie-deprecation-trial → enabled
chrome://flags/#tracking-protection-3pcd → enabled
chrome://flags/#tpcd-metadata-grants → disabled
chrome://flags/#tpcd-heuristics-grants → disabled
```

**Comment by hoodjoshua** [comment]
After taking a closer look at your demo page, the issue appears to be that after the script runs and injects the pay.google.com trial token, there is a pay.google.com iframe loaded without the trial token. 

The trial is persistent, so loading an iframe from the token origin without the trial token will disable the trial. This is noted under the graphic [here](https://developers.google.com/privacy-sandbox/3pcd/temporary-exceptions/third-party-deprecation-trial#inject_the_token_with_javascript).

**Comment by annamalai87** [comment]
You are right about the trial was not persistent as soon as the first iframe loaded without the token.
However after loading the iframe with the same token (By setting the origin-trial response header), still the trial seems to be disabled. Should we need to inject third party token in the top level site and load the iframe by setting its response header with another token (without third party matching)?

**Comment by hoodjoshua** [comment]
> Should we need to use two tokens? One for top level site with third party matching and another when loading the iframe?

You will need a different token for the iframe. The "third-party matching" tokens are only for use by scripts that inject the token on other origins. The tokens that can be used by iframes are provided for trial registrations where the "third-party matching" box is **not** checked.
